### DISCUSSION TITLE
Pass QerySet Manually and save result file

### DISCUSSION THREAD

**Discussion by Javokhir-1999** [django-import-export/django-import-export#1419] [Q&A] [answered]
Hi, I am doing a telegram bot. I used django and this import-export lib for admin. 
Now I have my QurySet result : 
**result** = Countries.objects.get(title = countryGlob)

I need to convert the **result** to Excel and save it. (than this file I will send to the user via telegram)
Is it possible to do this using `django-import-export`?. (manually without web interface)

I guess here I should use `workBook`  but I have the same problem with converting the `QerySet ` result to `workBook `
Thanks for your time!

**Comment by matthewhegarty** [comment, marked as answer]
> Is it possible to do this using django-import-export?. (manually without web interface)

Yes. it is possible.  Take a look at the [documentation](https://django-import-export.readthedocs.io/en/latest/getting_started.html) to get started.  you need to define a `Resource` object, then you call `export()` to get a tablib `Dataset`.  You can then [export this dataset to Excel](https://tablib.readthedocs.io/en/stable/formats.html#xlsx).

I suggest posting further questions to [StackOverflow](https://stackoverflow.com/questions/tagged/django-import-export)

For example:

```python
class BookResource(resources.ModelResource):

    class Meta:
        model = Book
        fields = ('id', 'author__name')


resource = BookResource()
dataset = resource.export()

with open('/tmp/output.xlsx', 'wb') as f:
    f.write(dataset.export('xlsx'))
```

**Reply by Javokhir-1999** [reply]
Ohh, Thanx a lot
